src/libs/DateUtils.js

```diff
@@ -115,21 +115,38 @@ function startCurrentDateUpdater() {
     });
 }
 
-/*
- * Updates user's timezone, if their timezone is set to automatic and
- * is different from current timezone
+/**
+ * @returns {Object}
  */
-function updateTimezone() {
+function getCurrentTimezone() {
     const currentTimezone = moment.tz.guess(true);
     if (timezone.automatic && timezone.selected !== currentTimezone) {
-        PersonalDetails.setPersonalDetails({timezone: {...timezone, selected: currentTimezone}});
+        return {...timezone, selected: currentTimezone};
     }
+    return timezone;
 }
 
 /*
- * Returns a version of updateTimezone function throttled by 5 minutes
+ * Updates user's timezone, if their timezone is set to automatic and
+ * is different from current timezone
+ */
+function updateTimezone() {
+    PersonalDetails.setPersonalDetails({timezone: getCurrentTimezone()});
+}
+
+// Used to throttle updates to the timezone when necessary
+let lastUpdatedTimezoneTime = moment();
+
+/**
+ * @returns {Boolean}
  */
-const throttledUpdateTimezone = _.throttle(() => updateTimezone(), 1000 * 60 * 5);
+function canUpdateTimezone() {
+    return lastUpdatedTimezoneTime.isBefore(moment().subtract(5, 'minutes'));
+}
+
```

Comment:
That could work. I'm not sure why we chose every 5 minutes or whether it would be easier to check if the timezone has changed. I think there is some kind of cost associated with trying to "guess" the timezone and maybe that's why this is throttled, but not really focused on improving this particular feature.